crates/ruff_linter/src/rules/pylint/rules/unnecessary_dunder_call.rs

```diff
@@ -156,8 +157,45 @@ pub(crate) fn unnecessary_dunder_call(checker: &mut Checker, call: &ast::ExprCal
         call.range(),
     );
 
-    if let Some(fixed) = fixed {
-        diagnostic.set_fix(Fix::safe_edit(Edit::range_replacement(fixed, call.range())));
+    if let Some(mut fixed) = fixed {
```

Comment:
I wonder if the fix produces invalid code if you have

```python
4.__sub__(
    3 
    + 
    4
)
```

What i understand is that we fix this to

```python
a -
    3
    +
    4
```

which Python doesn't like very much